src/MouseAdapter.js

```diff
@@ -309,20 +352,231 @@ define(function(require) {
+                // Check the 5 data points prior to those and determine if they were decreasing
+                var secondRange = this._getDeltaArrayIndex(-10);
+                if (this._calculateDeltaDirection(array,secondRange,5,this.DELTA_ARRAY_SIZE) === -1) {
+                    return true;
+                }
+            }
+            return false;
+        },
+
+        /**
+         * This is a helper function for detecting user scroll events.  When the user
+         * performs a scroll (at least with a track pad), the scroll delta drops to 
+         * nearly 0 before the new momentum is added to the scroll inertia. Because the 
+         * inertia typically decays very slowly over time, a significant decrease in the 
+         * scroll delta is a strong indication of a user event. This function works by 
+         * detecting these negative spikes in the scroll delta.
+         *
+         * One advantage this method has is that it can detect events as soon as they 
+         * occur because it does not take time to collect data.
+         * One disadvantage this method has is that it cannot detect events when the 
+         * the current inertia is below a certain threshold, including really light 
+         * scrolls and scrolls made after the current scroll inertia has decayed a lot.
+         * @param {Array.<number>} array The array of data to analyze for this check
+         * @return {boolean} Returns true if conditions were satisfied to indicate the
+         * user performed a scroll event, else false
+         * @private
+         */
+        _detectNegativeSpike: function(array) {
+            var lastPoint = this._getDeltaArrayIndex(-1);
+            // Check if the current event delta is low.  Nearly all of the negative spikes 
+            // seem to reach between 0.5-2.5, so make sure the point is low enough
+            if (array[lastPoint] <= 5) {
+                var recentPoint2 = this._getDeltaArrayIndex(-2);
+                var recentPoint3 = this._getDeltaArrayIndex(-3);
+                // Check to see if the previous two points are significantly higher.
+                // Checks last TWO points instead of just last one because the deltas
+                // seem to be prone to intermittent noisy spikes which can create false
+                // positives.
+                var isFirstPointHigherMagnitude = array[recentPoint2] / array[lastPoint] > 2;
+                var isSecondPointHigherMagnitude = array[recentPoint3] / array[lastPoint] > 2;
+                var isFirstPointHigherValue = array[recentPoint2] - array[lastPoint] > 2;
+                var isSecondPointHigherValue = array[recentPoint3] - array[lastPoint] > 2;
+                if ((isFirstPointHigherMagnitude && isFirstPointHigherValue) &&
+                    (isSecondPointHigherMagnitude && isSecondPointHigherValue)) {
+                    return true;
```

Comment:
What do you mean by this?  Like unit tests?  I dont think it's possible to test any of the conditions individually, as the conditions are only met if all are true.  Testing individual conditions wouldnt make any sense.